File: client/javascript-closure-angular/API/Client/Study.js
goog.provide('API.Client.Study');

/**
 * @record
 */
API.Client.Study = function() {}

/**
 * Sponsor key that this study belongs to
 * @type {!string}
 * @export
 */
API.Client.Study.prototype.sponsorKey;

/**
 * Unique study key
 * @type {!string}
 * @export
 */
API.Client.Study.prototype.studyKey;

/**
 * Mednet study ID (internal numeric identifier)
 * @type {!number}
 * @export
 */
API.Client.Study.prototype.studyId;

/**
 * Name of the study
 * @type {!string}
 * @export
 */
API.Client.Study.prototype.studyName;

/**
 * Description of the study
 * @type {!string}
 * @export
 */
API.Client.Study.prototype.studyDescription;

/**
 * Type of study (e.g., STUDY)
 * @type {!string}
 * @export
 */
API.Client.Study.prototype.studyType;

/**
 * Date when the study record was created
 * @type {!string}
 * @export
 */
API.Client.Study.prototype.dateCreated;

/**
 * Date when the study record was last modified
 * @type {!string}
 * @export
 */
API.Client.Study.prototype.dateModified;

File: client/scalaz/src/main/scala/org/openapitools/client/api/Study.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import Study._

case class Study (
  /* Sponsor key that this study belongs to */
  sponsorKey: Option[String],
/* Unique study key */
  studyKey: Option[String],
/* Mednet study ID (internal numeric identifier) */
  studyId: Option[Integer],
/* Name of the study */
  studyName: Option[String],
/* Description of the study */
  studyDescription: Option[String],
/* Type of study (e.g., STUDY) */
  studyType: Option[String],
/* Date when the study record was created */
  dateCreated: Option[String],
/* Date when the study record was last modified */
  dateModified: Option[String])

object Study {
  import DateTimeCodecs._

  implicit val StudyCodecJson: CodecJson[Study] = CodecJson.derive[Study]
  implicit val StudyDecoder: EntityDecoder[Study] = jsonOf[Study]
  implicit val StudyEncoder: EntityEncoder[Study] = jsonEncoderOf[Study]
}
